source/common/quic/quic_filter_manager_connection_impl.cc

```diff
@@ -169,6 +170,34 @@ void QuicFilterManagerConnectionImpl::onConnectionCloseEvent(
     ASSERT(network_connection_ != nullptr);
     network_connection_ = nullptr;
   }
+
+  if (!codec_stats_.has_value()) {
+    // The connection is closed before it can be used. stats are not recorded.
+    return;
+  }
+  switch (version) {
+  case quic::QUIC_VERSION_43:
+    codec_stats_->quic_version_43_.inc();
+    return;
+  case quic::QUIC_VERSION_46:
+    codec_stats_->quic_version_46_.inc();
+    return;
+  case quic::QUIC_VERSION_50:
+    codec_stats_->quic_version_50_.inc();
+    return;
+  case quic::QUIC_VERSION_51:
+    std::cout << "logging" << std::endl;
+    codec_stats_->quic_version_51_.inc();
+    return;
+  case quic::QUIC_VERSION_IETF_DRAFT_29:
+    codec_stats_->quic_version_h3_29_.inc();
+    return;
+  case quic::QUIC_VERSION_IETF_RFC_V1:
+    codec_stats_->quic_version_rfc_v1_.inc();
+    return;
+  default:
+    return;
```

Comment:
can we make sure we have unit tests for all of these for coverage?

Reply:
Sure. I added a new EnvoyClientSessionAllQuicVersionTest for this since none of the existing tests cover all QUIC versions.